stellargraph/mapper/sequences.py

```diff
@@ -527,3 +530,128 @@ def __len__(self):
+            be normalized or not. The default is True.
+        batch_size (int, optional): The batch size. It defaults to 1.
+        name (str, optional): An optional name for this generator object.
+    """
+
+    def __init__(
+        self, graphs, targets=None, normalize_adj=True, batch_size=1, name=None
+    ):
+
+        self.name = name
+        self.graphs = graphs
+        self.normalize_adj = normalize_adj
+        self.targets = targets
+        self.batch_size = batch_size
+        # we assume that all graphs have node features of the same dimensionality
+        self.node_features_size = graphs[0].node_features(graphs[0].nodes()).shape[1]
+
+        if targets is not None:
+            if len(graphs) != len(targets):
+                raise ValueError(
+                    "When passed together targets and the number of graphs should be the same length."
+                )
+
+            self.targets = np.asanyarray(targets)
+
+        self.on_epoch_end()
+
+    def __len__(self):
+        return int(np.ceil(len(self.graphs) / self.batch_size))
+
+    def __normalize_adj(self, adj):
+        adj = adj.tolil()
+        adj.setdiag(1)  # add self loops
+        degree_matrix_diag = 1.0 / np.sqrt(adj.sum(axis=1))
+        degree_matrix_diag = np.squeeze(np.asarray(degree_matrix_diag))
+        degree_matrix = sparse.lil_matrix(adj.shape)
+        degree_matrix.setdiag(degree_matrix_diag)
+        degree_matrix = degree_matrix.tocsr()
+        adj = degree_matrix @ adj @ degree_matrix
+        return adj
+
+    def __getitem__(self, index):
+        graphs = self.graphs[
+            index * self.batch_size : (index * self.batch_size) + self.batch_size
+        ]
+        adj_graphs = [graph.to_adjacency_matrix() for graph in graphs]
+
+        # The number of nodes for the largest graph in the batch. We are going to pad with 0 rows and columns
+        # the adjacency and node feature matrices (only the rows in this case) to equal in size the adjacency and
+        # feature matrices of the largest graph.
+        max_nodes = max([graph.number_of_nodes() for graph in graphs])
+
+        # The operations to normalize the adjacency matrix are a bit slow.
+        # FIXME: Either optimize this or implement as a layer.
+        if self.normalize_adj:
+            adj_graphs = [self.__normalize_adj(adj).toarray() for adj in adj_graphs]
+
+        graph_targets = None
+        #
+        if self.targets is not None:
+            graph_targets = self.targets[
+                index * self.batch_size : (index * self.batch_size) + self.batch_size
+            ]
+
+        # pad adjacency and feature matrices to equal the size of those from the largest graph
+        features = [
+            np.pad(
+                graph.node_features(graph.nodes()),
+                pad_width=(0, max_nodes - graph.number_of_nodes()),
+                mode="constant",
+                constant_values=0.0,
+            )[:, : self.node_features_size]
```

Comment:
I think this pads the second axis and then immediately strips it off?

Based on the `np.pad` docs, it looks like passing `(0, 0)` into `pad_width` should avoid doing any padding on that axis: `pad_width=((0, max_nodes - graph.number_of_nodes()), (0, 0)),` plus

```suggestion
            )
```

Reply:
I must have misunderstood the documentation because I could not get it to pad rows only. I'll have another look.